.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+
+## General Code Guidelines
+
+These guidelines apply to code that is written in AliceOS.
+
+### Follow the AliceOS API Styling
+
+Although AliceOS is a Python-based framework, our APIs are similarly written to those of the APIs used when creating macOS and iOS apps and are in camel case. This usually serves two purposes:
+
+- To identify AliceOS code from the rest of a visual novel project
+- To make it easy to understand when migrating from Apple platform development
+
+> Example: `applicationWillRequestNotification` vs. `application_will_request_notification`
+
+System-wide variables such as release names should be capitalized and use snake case as normal.
+
+> Example: `AS_SYS_VERSION` vs. `asSysVersion`
+
+### Follow commit conventions
+
+The AliceOS team has adopted the same [commit conventions from Clarity Moe](https://github.com/ClarityMoe/Community/blob/master/CHAPTER3-COMMIT%20CONVENTIONS.md), the parent organization running Sayonika. This commit style definitely explains where changes occured and what was changed in a succinct manner. However, there are some changes we have made to this commit style:
+
+- Common directories such as System, Applications, and Library do _not_ need to be fully typed out.
+- File extensions for frameworks (`.aosframework`), core services (`.aoscservice`), and applications (`.aosapp`) do _not_ need to be typed out.
+
+> Example: `[S/F/ServiceKit] Resolve misnaming of serviceWillRequestNotification from applicationWillRequestNotification`
+
+### Keep code organized
+
+The AliceOS structure is organized based on a simple macOS file structure to make everything easy to locate. Please try to keep this organization when contributing your code.
+
+## Issues
+
+These guidelines apply to issues on GitHub.
+
+### Be as descriptive and concise as possible
+
+So that AliceOS contributors and developers can better understand what the issue or request may be, issue descriptions should be concise but also descrptive. Refrain from writing an issue in a convoluted way that confuses others.
+
+Additionally, if you feel using a screenshot or video will better illustrate your description, add them in conjuction with (or to replace) the description. Remember to consult the [Screenshot Guidelines](#screenshots).
+
+### Label your issue during creation
+
+Issues are categorized by types such as `bug`, `enhancement`, `question`, etc. by contributors that can access labels. Since it isn't possible to tag an issue during creation, prepend the tag to your issue's title.
+
+> Example: [Bug] applicationShouldRequestNotification always returns False
+
+## Pull Requests
+
+These guides apply to pull requests on GitHub.
+
+### Describe all of your changes
+
+Pull requests generally include many changes that address a particular problem or a set of problems. Explain all of the changes you made; you may also specify the directories by using the commit message style.
```

Comment:
We define directory as a "subsystem" per the Linux commmit conventions. so granular changes are per-directory because its a subsystem.

Reply:
Clarified and put directories in parentheses for those that are unfamiliar.